### PR TITLE
Mat 6199/show definitions used section for edit test case page

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
## MADiE PR

Jira Ticket: [MAT-6199](https://jira.cms.gov/browse/MAT-6199)
(Optional) Related Tickets:

### Summary

- show definitions used section for edit test case page
- this also has changes from https://github.com/MeasureAuthoringTool/madie-patient/pull/538

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/editTestCase/groupCoverage/DefinitionsUsedSection.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+          fontFamily: "Rubik",
+          fontSize: "14px",
+          fontWeight: "500",
+          color: "#0073C8",
+        }}
+      >
+        Definition(s) Used
+      </div>
+      <div
+        style={{
```
Comment (adongare) [inline src/components/editTestCase/groupCoverage/DefinitionsUsedSection.tsx]: let's check with Michael about this. we may not need this style

File: src/components/editTestCase/groupCoverage/DefinitionsUsedSection.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import React from "react";
+import parse from "html-react-parser";
+
+const DefinitionsUsedSection = ({
+  results,
```
Comment (adongare) [inline src/components/editTestCase/groupCoverage/DefinitionsUsedSection.tsx]: receive single definition instead.

File: src/components/editTestCase/groupCoverage/DefinitionsUsedSection.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  cqlDefinitionCallstack,
+  groupCoverageResult,
+}) => {
+  const generateCallstackText = (): string => {
+    if (cqlDefinitionCallstack && groupCoverageResult) {
+      let text = "";
+      cqlDefinitionCallstack[results[0].name]?.forEach((calledDefinition) => {
+        // Get Highlighted HTML from execution results
+        text += groupCoverageResult.filter(
```
Comment (adongare) [inline src/components/editTestCase/groupCoverage/DefinitionsUsedSection.tsx]: `.find` is better choice here

File: src/components/editTestCase/groupCoverage/DefinitionsUsedSection.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+          (result) => result.name === calledDefinition.name
+        )[0].html;
+        getCallstack(calledDefinition.id).forEach((name) => {
```
Comment (adongare) [inline src/components/editTestCase/groupCoverage/DefinitionsUsedSection.tsx]: looks like a classic example of recursion?

File: src/components/editTestCase/groupCoverage/DefinitionsUsedSection.tsx (hunk 2 above)
Comment (sb-prateekkeerthi) [reply src/components/editTestCase/groupCoverage/DefinitionsUsedSection.tsx]: thank you, I'll change it to result instead

File: src/components/editTestCase/groupCoverage/DefinitionsUsedSection.tsx (hunk 3 above)
Comment (sb-prateekkeerthi) [reply src/components/editTestCase/groupCoverage/DefinitionsUsedSection.tsx]: sure, i'll change to find

File: src/components/editTestCase/groupCoverage/DefinitionsUsedSection.tsx (hunk 4 above)
Comment (sb-prateekkeerthi) [reply src/components/editTestCase/groupCoverage/DefinitionsUsedSection.tsx]: Yes, this is an example of recursion

File: src/components/editTestCase/groupCoverage/DefinitionsUsedSection.tsx (hunk 1 above)
Comment (sb-prateekkeerthi) [reply src/components/editTestCase/groupCoverage/DefinitionsUsedSection.tsx]: sure, I'll ask him

**Review by jkotanchik-SB — DISMISSED** [review]

**Review by adongare — APPROVED** [review]

**Pull request merged**